mambaSharedFramework/HLSValueTypes.swift

```diff
@@ -130,6 +131,92 @@ public func ==(lhs: HLSEncryptionMethodType, rhs: HLSEncryptionMethodType) -> Bo
+        /// Indicates that the Variant Stream could fail to play unless the output is protected by High-bandwidth
+        /// Digital Content Protection (HDCP) Type 0 or equivalent.
+        case type0 = "TYPE-0"
+        /// Indicates that the Variant Stream could fail to play unless the output is protected by HDCP Type 1 or
+        /// equivalent.
+        case type1 = "TYPE-1"
+    }
+    public init?(string: String) {
+        self.init(hdcpLevel: string)
+    }
+    public init?(hdcpLevel: String) {
+        guard let type = HDCPLevel(rawValue: hdcpLevel) else {
+            return nil
+        }
+        self.type = type
+    }
+    public init(hdcpLevel: HDCPLevel) {
+        self.type = hdcpLevel
+    }
+}
+
+/// Represents the dynamic range of the video.
+///
+/// This is represented by an enumeration where each case covers a group of similar opto-electronic transfer
+/// characteristic functions that could have been used to encode the media file.
+///
+/// For example, `SDR` covers TransferCharacteristics code points 1, 6, 13, 14 and 15. More information on what each
+/// code point represents can be found in _"Information technology - MPEG systems technologies - Part 8: Coding-_
+/// _independent code points" ISO/IEC International Standard 23001-8, 2016_ [CICP].
+public struct HLSVideoRange: Equatable, FailableStringLiteralConvertible {
+    public let type: VideoRange
+    public enum VideoRange: String {
+        /// The value MUST be SDR if the video in the Variant Stream is encoded using one of the following reference
+        /// opto-electronic transfer characteristic functions specified by the TransferCharacteristics code point: 1, 6,
+        /// 13, 14, 15. Note that different TransferCharacteristics code points can use the same transfer function.
+        case sdr = "SDR"
+        /// The value MUST be HLG if the video in the Variant Stream is encoded using a reference opto-electronic
+        /// transfer characteristic function specified by the TransferCharacteristics code point 18, or consists of such
+        /// video mixed with video qualifying as SDR.
+        case hlg = "HLG"
+        /// The value MUST be PQ if the video in the Variant Stream is encoded using a reference opto-electronic
+        /// transfer characteristic function specified by the TransferCharacteristics code point 16, or consists of such
+        /// video mixed with video qualifying as SDR or HLG.
+        case pq = "PQ"
+    }
+    public init?(string: String) {
+        self.init(videoRange: string)
+    }
+    public init?(videoRange: String) {
+        guard let type = VideoRange(rawValue: videoRange) else {
+            return nil
+        }
+        self.type = type
+    }
+    public init(videoRange: VideoRange) {
+        self.type = videoRange
+    }
+}
+
+/// Represents the format of the file referenced by `EXT-X-SESSION-DATA:URI`.
+public struct HLSSessionDataFormat: Equatable, FailableStringLiteralConvertible {
```

Comment:
Something I was never quite clear on is why do we have these `struct`'s that are basically a wrapper around a property backed by an enum. In other words, why can't we just have this?

```swift
public enum HLSSessionDataFormatType: String, Equatable, FailableStringLiteralConvertible {
    case json = "JSON"
    case raw = "RAW"
    
    public init?(string: String) {
        guard let type = HLSSessionDataFormatType(rawValue: string) else {
            return nil
        }
        self = type
    }
}
```

Reply:
Yeah, no idea, I always had the same thoughts. I'll make a change to favor this style... My only concern is that we will have a divergence of styles between other types... But maybe that's OK 🤷

Reply:
Updated here: 34de4a993d2206f793afe18e53db83b07d7f8ca8